aequilibrae/project/zone.py

```diff
@@ -75,8 +82,22 @@ def add_centroid(self, point: Point) -> None:
         if point is None:
             point = self.geometry.centroid
 
+        if robust:
+            check_sql = """SELECT count(*) FROM nodes
+                             WHERE  nodes.geometry = GeomFromWKB(?, 4326) AND
+                          nodes.ROWID IN (
+                           SELECT ROWID FROM SpatialIndex WHERE f_table_name = 'nodes' AND
+                           search_frame = GeomFromWKB(?, 4326))
+                       """
+
+            test_list = self.conn.execute(check_sql, [point.wkb, point.wkb]).fetchone()
+            while sum(test_list):
+                test_list = self.conn.execute(check_sql, [point.wkb, point.wkb]).fetchone()
+                point = Point(point.x + random.random() * shift,
+                              point.y + random.random() * shift)
```

Comment:
I would consider extracting this to a method `perturb_point_to_avoid_existing_nodes`
Then you could test it in isolation from the insert to database - and this method would become 

```
if robust:
    point = perturb_point_to_avoid_existing_nodes(point)
```

Reply:
Yeah... It would've been better...